### ISSUE TITLE
TVGuide timer BUG

### ISSUE THREAD

**Issue by marboroman** [E2OpenPlugins/e2openplugin-OpenWebif#588]
when setting the timer in tv guide the timer sign appears in the wrong place. My screen res is 1920*1200. The video of the bug: http://radikal.ru/video/GXl3T08BPVw

**Comment by rdamas** [comment]
Strange and not reproducible here. Do you have experience with debugging Javascript and/or HTML from the browser? Can you inspect the HTML blocks "<div class="event"...data-id="nnn">" if there are non unique data-id's for the events in question?

**Comment by marboroman** [comment]
can г try to reproduce it with the same screen resolution (1920*1200)? It's just a visualization bug: if I refresh the page and check the wrong timer it appears that it's correct.

**Comment by rdamas** [comment]
It has nothing to do with screen resolution; responsible is the following Javascript function:
'''function cbAddTimerEvent(state) {
        if (state.state) {
                $('.event[data-id='+state.eventId+'] .timer').remove();
                $('.event[data-id='+state.eventId+'] div:first').append('<div class="timer">'+tstr_timer+'</div>');
        }
}
'''
which searches the event with data-id=state.eventId. state.eventId holds the event-id of the block you selected to add as timer, and is called via onlick(), e.g.:
'''<a class="tb" href="#" onclick="addTimerEvent('1:0:19:2B8E:3F2:1:C00000:0:0:0:',24651,false,cbAddTimerEvent);return false;"
'''
Here 24651 is the event id, and I suspect that id is not unique. Or the browser has a horrible bug, which I don't believe.

**Comment by marboroman** [comment]
i tried firefox and chrome and both have the same bug.

**Comment by Dima73** [comment]
most likely event id is not unique

**Comment by marboroman** [comment]
yes, i compared the id's and when it sets the timer to the wrong place that place has the same id. Is it because of the faulty epg source? I use the remote xmltv file.

**Comment by rdamas** [comment]
Likely, yes.

**Comment by marboroman** [comment]
ok. will try to switch to different source. Thank you.
